### ISSUE TITLE
No option to clean up phone storage

### ISSUE THREAD

**Issue by ahendre2010** [nextcloud/android#1146]
### Actual behaviour
-Files do get backed up on server with keep source option in auto upload mode.
-If user wanted to free up storage by deleting backed up files, there is no quick way.

### Expected behaviour
-There could be an option in settings menu to free up phone storage.
-This option when clicked checks files which are backed up on to a server and removes them from phone storage.
 
Nextcloud app version: v.1.4.3

**Comment by tobiasKaminsky** [comment]
If I understand you correct you want to have, e.g. images that are uploaded to NC, but remain in DCIM folder to be deleted as the app can verify that those are already uploaded?
This is currently not possible and I guess hard to do as we then would have to track every link between uploaded file and source file.
I think the option "move to nextcloud folder" is helping to solve this. You upload & move an image. If you need more space you can just delete the instant upload folder.
(from your email this morning I thought this is about to "just" delete all downloaded files)

**Comment by ahendre2010** [comment]
Yes I think what you are saying is a viable option. But it took me a bit to understand what exactly you are saying.
Anyway I would like to tell you I am not a experienced developer but have some knowledge of programming for hobby projects.
I was thinking about some index file which is a replica of a file structure on NC and phone can be used to compare, can it be useful ?

**Comment by tobiasKaminsky** [comment]
An index to compare between NC and file structure is something we somehow need for real 2-way sync, but this we need to plan in a real proper way and we started in #285

**Comment by jasonbayton** [comment]
I've seen a few mainstream apps start to offer this, so expect it's not outside the realm of possibility! I still think a client-server checksum would be the ultimate solution though.

**Comment by jtagcat** [comment]
See also: https://github.com/nextcloud/ios/issues/1240
